docs/source/development/defining_new_models.md

```diff
@@ -0,0 +1,233 @@
+# How to add a new model to the Virtual Rainforest
+
+The Virtual Rainforest is designed to be modular, meaning that the set of models to be
+used in a particular run is configurable at the start of the simulation. We are starting
+out by defining a core set of models (`abiotic`, `animals`, `plants` and `soil`), which
+will generally all be included for the vast majority of simulations. In the future, it
+might be desirable to include models for other aspects of rainforests (e.g.
+`freshwater`, `disturbance`), or to include multiple modelling approaches for a process.
+When this happens a new model should be created. This page will set out the process for
+adding a new model to the Virtual Rainforest in a manner that allows it to be used
+appropriately by the `core` simulation functionality.
+
+## Define a new `Model` class
+
+You should first start by defining a new folder for your model (within
+`virtual_rainforest/models/`).
+
+```bash
+mkdir virtual_rainforest/models/freshwater
+```
+
+Within this folder a `python` script defining the model should be created.
+
+```bash
+touch virtual_rainforest/models/freshwater/model.py
+```
+
+This script must import a number of things to be able to set up a new `Model` class
+correctly. It's worth noting that `log_and_raise` will be removed soon.
+
+```python
+# One of the member functions of the Model class returns a class instance. mypy doesn't
+# know how to handle this unless annotations are imported from __future__
+from __future__ import annotations
+
+# Any needed for type hints of the config dictionary as the values are of various types
```

Comment:
I believe you can simply quote the return name to avoid the `mypy` error too. We should do this systematically though.

Reply:
Didn't realise that worked! Prefer this way, partly because its a little neater, partly because I like importing from `__future__`